lib/kitchen/driver/ssh_base.rb

```diff
@@ -115,6 +115,14 @@ def verify(state) # rubocop:disable Metrics/AbcSize
       rescue Kitchen::Transport::TransportFailed => ex
         raise ActionFailed, ex.message
       ensure
+        if verifier[:scp_file]
```

Comment:
Concern are very valid.

However I'm not sure how to proceed. My use case is the use of shell verifier( https://github.com/test-kitchen/test-kitchen/pull/741) which runs wrapper around serverspec test suite. 

If the wrapper returns non-zero exit status it raises ssh TransportFailed ...
This behavior aslo relates to ssh transport design.

So in fact I'm not sure how to proceed with moving it out of `ensure` and avoid shotgun surgery of ssh transport.

Do you have any idea ?